contract/services.py

```diff
@@ -351,6 +354,50 @@ def __copy_details(self, contract_id, amended_contract):
             cd_new.json_ext = None
             cd_new.save(username=self.user.username)
 
+    @check_authentication
+    def renew(self, contract):
+        try:
+            # check rights for delete contract
+            if not self.user.has_perms(ContractConfig.gql_mutation_renew_contract_perms):
+                raise PermissionError("Unauthorized")
+            from core import datetime, datetimedelta
+            contract_to_renew = ContractModel.objects.filter(id=contract["id"]).first()
+            contract_id = contract["id"]
+            # block deleting contract not in Updateable or Approvable state
+            state_right = self.__check_rights_by_status(contract_to_renew.state)
+            # check if we can amend
```

Comment:
please update the comment

Reply:
ok, I will update all coments here with wrong words not related to that service